src/bin/ion/commands/beta/generate/utils.rs

```diff
@@ -88,7 +88,7 @@ impl Language for JavaLanguage {
     }
 
     fn is_built_in_type(name: &str) -> bool {
-        matches!(name, "int" | "String" | "boolean" | "byte[]" | "float")
+        matches!(name, "int" | "String" | "boolean" | "byte[]" | "double")
```

Comment:
(🗺️ PR tour) ion-java uses [`doubleValue`](https://javadoc.io/doc/software.amazon.ion/ion-java/1.0.0/software/amazon/ion/IonReader.html) API for `IonReader`, so changing this similar to ion-java.